source/constructing-lattices.md

```diff
@@ -0,0 +1,351 @@
+C, B, A, C, B, A, C, B, A
+```
+
+Notice that reverse order does not mean true [direction reversal](#s:ref.construct). 
+For elements that have longitudinal symmetry, this does not matter. 
+However, for example, for a `Bend` element that is in a line with reversed order,
+the edge angle `e1` will still represent the edge of the upstream side and `e2` will represent the edge 
+at the downstream side.
+
+%---------------------------------------------------------------------------------------------------
+(s:direction)=
+### Direction reversal
+
+The optional `direction` component of `item` can be used for true [direction reversal](#s:ref.construct).
+Possible values are `+1` and `-1`. The Default is `+1` which represents an unreversed element
+or BeamLine. BeamLine reversal involves both reversed order of the line and direction reversal of
+the individual line items. Example:
+```{code} yaml
+BeamLine:
+  name: lineA
+  line:
+    - item: 
+        direction: -1
+        name: lineB
+
+BeamLine:
+  name: lineB
+  line:
+    - item: ele1
+    - item:
+        direction: -1
+        name: ele2
+```
+The expanded `lineA` has elements:
+```{code} yaml
+ele2, -ele1
+```
+where the negative sign here indicates that `ele1` is reversed in direction. 
+Notice that `ele2` is not reversed since the reversal of a reversed element results
+in an element that is unreversed.
+
+%---------------------------------------------------------------------------------------------------
+(s:placement)=
+### Line item placement
+
+```{figure} figures/superimpose.svg
+:width: 80%
+:name: f:superposition
+
+Positioning of a line item (which may be a lattice element or BeamLine) with respect to 
+a reference line item when there is an explicit placement component present. Assuming
+unreversed elements, a positive offset positions the item being positioned downstream
+of the reference item.
+
+```
+
+By default,
+a line `item` is placed such that the entrance end of the `item` is flush with the exit end
+of the preceding `item` as explained in the [Branch Coordinates Construction](#s:ref.construct) section.
+To adjust the longitudinal placement of an `item`, 
+the `placement` component of an `item` can be used.
+When there is a `placement` component, figure {numref}`f:superposition` shows how the line `item` 
+is positioned with respect to a reference line `item`. 
+If the reference `item` is not specified, the beginning of the `line` is used. In this case, 
+a `reference_origin` may not be specified.
+
+The components of `placement` are:
+```{code} yaml
+offset            # Optional Real [m]. Longitudinal offset of the line item. Default is zero.
+origin            # Optional switch. Line item origin point. Default is ENTRANCE_END.
+reference         # Optional string. Reference line item. Default is a blank string which 
+                  #  indicates the beginning of the `line`.
+reference_origin  # Optional switch. Reference line item origin point. Default is EXIT_END.
```

Comment:
Having both a `reference_point` (above) and `reference_origin` will be confusing, especially since the `reference_origin` is not the origin of one of the previously-defined coordinate systems.  In general, I think the word `reference` is a bit overloaded.  What about `offset_marker` or `offset_point`, or something similar?

Reply:
I have changed some names. Please have a look.